python/GangaLHCb/Lib/Tasks/LHCbTransform.py

```diff
@@ -117,38 +117,26 @@ def createUnits(self):
                     assigned_data.files += unit.inputdata.files
 
                 # any new files
-                new_data = LHCbDataset( files = self.inputdata[id].difference(assigned_data).files )
+                new_data = LHCbDataset(files=self.inputdata[id].difference(assigned_data).files)
 
                 if len(new_data.files) == 0:
                     continue
 
-                # create units for these files
-                if self.files_per_unit > 0:
-
-                    # loop over the file array and create units for each set
-                    num = 0
-                    while num < len(new_data.files):
-                        unit = LHCbUnit()
-                        unit.name = "Unit %d" % len(self.units)
-                        unit.input_datset_index = id
-                        self.addUnitToTRF(unit)
-                        unit.inputdata = copy.deepcopy(self.inputdata[id])
-                        unit.inputdata.files = []
-                        unit.inputdata.files += new_data.files[
-                            num:num + self.files_per_unit]
-                        num += self.files_per_unit
-
-                else:
-                    # new unit required for this dataset
+                # Create units for these files
+                step = self.files_per_unit
+                if step <= 0:
+                    step = int(1e100)
```

Comment:
I'm not sure I like setting this to an arbitrarily large size. Would it not be better to set it to `len(new_data.files)`?